contracts/SpokePool.sol

```diff
@@ -885,7 +885,11 @@ abstract contract SpokePool is
     ) public override nonReentrant unpausedFills {
         // Exclusivity deadline is inclusive and is the latest timestamp that the exclusive relayer has sole right
         // to fill the relay.
-        if (relayData.exclusivityDeadline >= getCurrentTime() && relayData.exclusiveRelayer != msg.sender) {
+        if (
+            relayData.exclusiveRelayer != msg.sender &&
+            relayData.exclusivityDeadline >= getCurrentTime() &&
+            relayData.exclusiveRelayer != address(0)
+        ) {
             revert NotExclusiveRelayer();
         }
```

Comment:
We now have this check in 3 places; worth factoring out to a function?

Reply:
I factored out the common logic to a `_fillIsExclusive()` function. I think this is probably the right call but could be persuaded otherwise.